src/MPP/mpp_land.F

```diff
@@ -2321,500 +2438,501 @@ end subroutine sum_real8
-                      call mpi_recv(tmp,nsize,&
-                              MPI_INTEGER8,i,tag,HYDRO_COMM_WORLD,mpp_status,ierr)
-                      do k = 1, nsize
-                          if(in0(k) .ne. tmp(k)) in(k) = tmp(k)
-                      end do
-                  end if
-              end do
-          end if
-          call mpp_land_bcast_int8_1d(in)
-
-      end subroutine updateLake_seqInt8
-
-
-      subroutine updateLake_seq(in,nsize,in0)
-       implicit none
-       integer :: nsize
-       real, dimension(nsize) :: in
-       real, dimension(nsize) :: tmp
-       real, dimension(:) :: in0
-       integer tag, i, status, ierr, k
-       if(nsize .le. 0) return
-
-       tag = 29
-       if(my_id .ne. IO_id) then
-          call mpi_send(in,nsize,MPI_REAL, IO_id,     &
-                tag,HYDRO_COMM_WORLD,ierr)
-       else
-          do i = 0, numprocs - 1
+         end do
+      end if
+      call mpp_land_bcast_int1d(in)
+
+   end subroutine updateLake_seqInt
+
+   subroutine updateLake_seqInt8(in,nsize,in0)
+      implicit none
+      integer :: nsize
+      integer(kind=int64), dimension(nsize) :: in
+      integer(kind=int64), dimension(nsize) :: tmp
+      integer(kind=int64), dimension(:) :: in0
+      integer tag, i, status, ierr, k
+      if(nsize .le. 0) return
+
+      tag = 29
+      if(my_id .ne. IO_id) then
+         call mpi_send(in,nsize,MPI_INTEGER8, IO_id,     &
+            tag,HYDRO_COMM_WORLD,ierr)
+      else
+         do i = 0, numprocs - 1
             if(i .ne. IO_id) then
                call mpi_recv(tmp,nsize,&
-                   MPI_REAL,i,tag,HYDRO_COMM_WORLD,mpp_status,ierr)
+                  MPI_INTEGER8,i,tag,HYDRO_COMM_WORLD,mpp_status,ierr)
                do k = 1, nsize
                   if(in0(k) .ne. tmp(k)) in(k) = tmp(k)
                end do
             end if
-          end do
-       end if
-       call mpp_land_bcast_real_1d(in)
-
-     end subroutine updateLake_seq
-
-
-     subroutine updateLake_seq_char(in,nsize,in0)
-       implicit none
-       integer :: nsize
-       character(len=256), dimension(nsize) :: in
-       character(len=256), dimension(nsize) :: tmp
-       character(len=256), dimension(:) :: in0
-       integer tag, i, status, ierr, k, in_len
-       if(nsize .le. 0) return
-
-       in_len = size(in, 1)*len(in)
-
-       tag = 29
-       if(my_id .ne. IO_id) then
-          call mpi_send(in,in_len,MPI_CHARACTER, IO_id,     &
-                tag,HYDRO_COMM_WORLD,ierr)
-       else
-          do i = 0, numprocs - 1
+         end do
+      end if
+      call mpp_land_bcast_int8_1d(in)
+
+   end subroutine updateLake_seqInt8
+
+
+   subroutine updateLake_seq(in,nsize,in0)
+      implicit none
+
+      integer :: nsize
+      real, dimension(:), intent(inout), asynchronous :: in
+      real, dimension(:), intent(in) :: in0
+
+      real, dimension(nsize) :: tmp
+      real, dimension(:), allocatable, asynchronous :: prev
```

Comment:
Did you find that the asynchronous attribute was needed here? I'm just curious because from what I can gather in [this thread](https://fortran-lang.discourse.group/t/when-to-use-the-asynchronous-attribute-and-why/109/5) it wouldn't make a difference in this case but maybe I'm missing something or interpreting things wrong. I think it's probably fine, just interested if you found that it made a difference.